pkg/cmd/roachtest/test_runner.go

```diff
@@ -1314,28 +1317,13 @@ func (r *testRunner) runTest(
 		// Actively poll for VM preemptions, so we can bail out of tests early and
 		// avoid situations where a test times out and the flake assignment logic fails.
 		monitorForPreemptedVMs(runCtx, t, c, l)
-		// This is the call to actually run the test.
-		s.Run(runCtx, t, c)
-	}()
 
-	// Monitor the task manager for completed events, or failure events and log
-	// them. A failure will call t.Errorf which cancels the test's context.
-	go func() {
-		for {
-			select {
-			case event := <-t.taskManager.CompletedEvents():
-				if event.Err == nil {
-					t.L().Printf("task finished: %s", event.Name)
-					continue
-				} else if event.TriggeredByTest {
-					t.L().Printf("task canceled by test: %s", event.Name)
-					continue
-				}
-				t.Errorf("task `%s` returned error: %v", event.Name, event.Err)
-			case <-runCtx.Done():
-				return
-			}
+		monitorTasks(runCtx, t.taskManager, t, l)
+		if t.spec.Monitor {
+			testMonitor.start()
```

Comment:
Double checking my understanding, it should be okay to start the monitor here before any cockroach processes have started because systemctl will report nothing running?